### PR TITLE
Feature/rub 537

### PR DESCRIPTION
**Pull request by CRMTravis**

### CODE REVIEW

File: force-app/main/default/aura/FinancialVerificationDetailsComponent/FinancialVerificationDetailsComponentHelper.js [hunk 1]
```diff
@@ -127,6 +127,29 @@
...
+    let primarySourceOfincomeField;
+    let dataMap = component.get("v.ParentVerifiedDataMap");
+    let primarySourceOfIncome = dataMap.primarySourceOfIncomeVerified;
+    for (let key in primarySourceofIncomeMap) {
+      if (primarySourceOfIncome.includes(key)) {
+        primarySourceOfincomeField = primarySourceofIncomeMap[key];
+        break;
+      }
+    }
+    console.log("Primary Source of income: ", primarySourceOfIncome);
```
Comment (RemarioCRM) [inline force-app/main/default/aura/FinancialVerificationDetailsComponent/FinancialVerificationDetailsComponentHelper.js:141]: remove all console logs and dead code

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/FinancialDetailsSelector.cls [hunk 2]
```diff
@@ -92,7 +92,11 @@ public with sharing class FinancialDetailsSelector {
         Primary_Source_of_Income_VERIFIED_list__c,
```
Comment (RemarioCRM) [inline force-app/main/default/classes/FinancialDetailsSelector.cls:92]: ensure all queries have WITH SECURITY_ENFORCED

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/aura/FinancialVerificationDetailsComponent/FinancialVerificationDetailsComponentHelper.js (hunk 1 above)
Comment (CRMTravis) [reply force-app/main/default/aura/FinancialVerificationDetailsComponent/FinancialVerificationDetailsComponentHelper.js]: Where is the dead code?

File: force-app/main/default/aura/FinancialVerificationDetailsComponent/FinancialVerificationDetailsComponentHelper.js [hunk 3]
```diff
@@ -127,6 +127,20 @@
     component.set("v.UnverifiedDebts", unverifiedDebts);
   },
 
+  getSelectedPrimarySourceOfIncomeValue: function (component) {
```
Comment (RemarioCRM) [inline force-app/main/default/aura/FinancialVerificationDetailsComponent/FinancialVerificationDetailsComponentHelper.js:130]: add jsdocs

File: force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp [hunk 4]
```diff
@@ -11,10 +11,12 @@
 <aura:component controller="FinancialDetailsController">
   <aura:attribute name="toggleCheckAll" type="Boolean" default="false" />
   <aura:attribute name="auraIdList" type="List" />
-  <aura:attribute name="verifiedMonthlyIncomeMap" type="Map" />
-  <aura:attribute name="verifiedDataMap" type="Map" default="{}" />
+  <aura:attribute name="VerifiedTotalsMap" type="Map" />
+  <aura:attribute name="childVerifiedDataMap" type="Map" default="{}" />
```
Comment (RemarioCRM) [inline force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp:15]: why is this needed, **childVerifiedDataMap** all of a sudden

File: force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp [hunk 5]
```diff
@@ -23,8 +25,8 @@
   />
   <aura:handler
     name="change"
-    action="{!c.onVerifiedMonthlyIncomeMapChange}"
-    value="{!v.verifiedMonthlyIncomeMap}"
+    action="{!c.onChildVerifiedDataMapChange}"
```
Comment (RemarioCRM) [inline force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp:28]: whats the real difference between onVerifiedMonthlyIncomeMapChange vs onChildVerifiedDataMapChange

File: force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponentController.js [hunk 6]
```diff
@@ -97,10 +89,10 @@
     }
   },
 
-  onVerifiedMonthlyIncomeMapChange: function (component, event) {
-    let verifiedMonthlyIncomeData = component.get("v.verifiedMonthlyIncomeMap");
-    let verifiedData = component.get("v.verifiedDataMap");
-    let data = Object.assign(verifiedData, verifiedMonthlyIncomeData);
-    component.set("v.verifiedDataMap", data);
+  onChildVerifiedDataMapChange: function (component, event, helper) {
```
Comment (RemarioCRM) [inline force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponentController.js:92]: how do you prevent recursion?

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp (hunk 4 above)
Comment (CRMTravis) [reply force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp:15]: All of a sudden?

Comment (CRMTravis) [reply force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp:15]: Its been there from initial Implementation of the ticket which you removed.

File: force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp (hunk 5 above)
Comment (CRMTravis) [reply force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp:28]: Semantics

File: force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponentController.js (hunk 6 above)
Comment (CRMTravis) [reply force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponentController.js:92]: I made the Parent update a different map.

File: force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp (hunk 4 above)
Comment (RemarioCRM) [reply force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp:15]: childVerifiedDataMap is newly added?

File: force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp (hunk 5 above)
Comment (RemarioCRM) [reply force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponent.cmp:28]: it is not, most of your center around this change. I need to know why

File: force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponentController.js (hunk 6 above)
Comment (RemarioCRM) [reply force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponentController.js:92]: which ones?

Comment (CRMTravis) [reply force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponentController.js:92]: I parsed the Financial Data into 3 maps Verified, Unverified and Totals. Unverified Map details are load upon initialisation into the child components. When change occurs on the child component belonging to the Verified Data map then the parent listens to it and makes the necessary calculations. the result of calculations are stored in a separate map for totals only which the child listens to and updates the appropriated totals. Parent child does not handle the or listen for totals changes so recursion is avoided.

Comment (RemarioCRM) [reply force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponentController.js:92]: neat trick

Comment (CRMTravis) [reply force-app/main/default/aura/MonthlyIncomeVerificationLineComponent/MonthlyIncomeVerificationLineComponentController.js:92]: Thanks

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/staticresources/productDetailsGeneralFunctions.js [hunk 7]
```diff
@@ -1082,55 +1082,69 @@ window.financialVerificationComponentTotalsController = function (
...
+    }
+    if (calculationKey.includes("totalMonthlyExpensesVerified")) {
+      result = calculateTotalMonthlyExpenses(
+        calculationMap[calculationKey],
+        dataMap
+      );
+      totalsMap.totalMonthlyExpensesVerified = result;
+      return totalsMap;
+    }
+    if (calculationKey.includes("totalDebtConsolidatedVerified")) {
```
[the remark this answers is not in the document]
Comment (CRMTravis) [reply force-app/main/default/staticresources/productDetailsGeneralFunctions.js:1123]: Forgot that we had a js file for constants tbh

**Review by RemarioCRM — APPROVED** [review]

**Pull request merged**